markdown-service: update ordered lists

knholland opened the pull request:
An update is needed to add markdown processing to numbered lists that will exclude numbers not meant to be in a list.


Reviewed-By: Katie Owen <katherine.owen@turner.com>

knholland committed:
markdown-service: update ol

knholland committed:
Merge pull request #1 from khowen/update-list-support

markdown-service: update ol

knholland committed:
markdown-service: testing

knholland committed:
markdown-service: testing

knholland committed:
markdown-service: testing

knholland committed:
markdown-service: testingk

knholland committed:
markdown-service: testing

knholland committed:
updates: testing

jamsyoung commented on lib/markdown-service.js:
```diff
@@ -66,8 +66,11 @@ class MarkdownService {
+            // https://regex101.com/r/SO5haw/1
+            if (data.match(/^\d{1,4}\./)) {
+                // https://regex101.com/r/nB8bV3/4
+                data = data.replace(/(^\d{1,4})(.)/, '$1\\$2');
```
As it stands, this line will match any line that starts with one to four digits followed by any non-line-breaking character.

An example where we would not want to match would be with non-numeric identifiers, such as `1a.`.

We are specifically referring to this; CURRENT:

```
data = data.replace(/(^\d{1,4})(.)/, '$1\\$2');
```

versus this; UPDATED:

```
data = data.replace(/(^\d{1,4})(\.)/, '$1\\$2');
```

With the assumption that we do not want to match non-numeric identifiers, I added a test case to the regex101 test.

https://regex101.com/r/nB8bV3/5/tests

This shows the CURRENT pattern matching, which we do not want to do, so the test fails.

Updating the regex to the UPDATED version allows the test to pass.

This can be seen here - https://regex101.com/r/nB8bV3/6/tests


What do you think?

knholland replied:
The if statement with `data.match(/^\d{1,4}\./)` should keep any incoming data from hitting this regex unless it is a number followed by a ".". 

In testing, I cannot get `1a.` to enter into this if statement. (I added this if statement because in previous testing, I noticed it was possible to have numerals in places we wouldn't want to necessarily format. The intent of the if statement is to only allow 1-4 digit numberals followed by a "." to be processed using `data.replace(/(^\d{1,4})(.)/, '$1\\$2')`.)

We only want to format if it  is a numeral followed by a "." Right?

Is the intent to remove this if statement and have a standalone regex replace?
https://regex101.com/r/SO5haw/2/tests

jamsyoung replied:
Now that you mention it, yes, it would be better to do away with the conditional entirely since the body of the conditional is only one line.

jamsyoung commented on test/unit/markdown-service-test.js:
```diff
@@ -280,14 +280,63 @@ describe('Markdown Service', () => {
 
     it('should convert single digit numbered lists to add \\\\ between the digit and the digit\'s "."', () => {
```
Update description to match the updated test (2 slashes vs. 4)

knholland replied:
These descriptions show 4 slashes vs. 2 to ensure the user sees the correct number of slashes in the description when run `npm test`

With 2 slashes:
 ✓ should convert single digit numbered lists to add \\ between the digit and the digit's "."

With 4 slashes:
✓ should convert single digit numbered lists to add \\\\ between the digit and the digit's "."

jamsyoung replied:
Pesky slashes

jamsyoung commented on test/unit/markdown-service-test.js:
```diff
@@ -280,14 +280,63 @@ describe('Markdown Service', () => {
         response.should.equal(expectedResponse);
     });
 
     it('should convert double digit numbered lists to add \\\\ between the digit and the digit\'s "."', () => {
```
Update description to match the updated test (2 slashes vs. 4)

jamsyoung commented on test/unit/markdown-service-test.js:
```diff
@@ -280,14 +280,63 @@ describe('Markdown Service', () => {
+        response.should.equal(expectedResponse);
+    });
+
+    it('should convert a triple digit numbered lists to add \\\\ between the digit and the digit\'s "."', () => {
```
Update description to match the updated test (2 slashes vs. 4)

jamsyoung commented on test/unit/markdown-service-test.js:
```diff
@@ -280,14 +280,63 @@ describe('Markdown Service', () => {
+        response.should.equal(expectedResponse);
+    });
+
+    it('should convert a four digit numbered lists to add \\\\ between the digit and the digit\'s "."', () => {
```
Update description to match the updated test (2 slashes vs. 4)

knholland committed:
Merge branch 'develop' of https://github.com/cnnlabs/cnn-markdown-service into develop

knholland committed:
Merge branch 'develop' into update-numbered-lists

knholland committed:
example: add comma

A condition exists causing an error due to a missing comma on line 72;
an update is needed to add this comma.

knholland commented:
Updated to remove conflict in example.js.

zoitec reviewed (approved):
Looks good
